src/Groups/group_constructors.jl

```diff
@@ -475,27 +475,47 @@ projective_omega_group(n::Int, q::Int) = projective_omega_group(0, n, q)
 #
 ################################################################################
 
-"""
+@doc raw"""
     free_group(n::Int, s::VarName = :f; eltype::Symbol = :letter) -> FPGroup
-    free_group(L::Vector{<:VarName}) -> FPGroup
     free_group(L::VarName...) -> FPGroup
+    free_group(varnames_specifiers...) -> FPGroup
+
+Return a free group.
 
-The first form returns the free group of rank `n`, where the generators are
-printed as `s1`, `s2`, ..., the default being `f1`, `f2`, ...
-If `eltype` has the value `:syllable` then each element in the free group is
-internally represented by a vector of syllables,
-whereas a representation by a vector of integers is chosen in the default case
-of `eltype == :letter`.
+The first form returns a free group of rank `n`, where the generators are
+printed as `"$s1"`, `"$s2"`, ..., the default being `f1`, `f2`, ...
 
-The second form, if `L` has length `n`, returns the free group of rank `n`,
-where the `i`-th generator is printed as `L[i]`.
+The second form returns a free group of rank `n`, where `n` is the length of `L`,
+and `L` consists of strings, symbols or characters giving the variable names.
 
-The third form, if there are `n` arguments `L...`,
-returns the free group of rank `n`,
-where the `i`-th generator is printed as `L[i]`.
+In the final form, the argument list consists of a sequence of one or
+more of the following:
+1. A vector `L` of variable names.
+2. A pair of the form `A => B`, where `A` is a `VarName` (so a string, symbol
+   or character) and `B` is a range or more generally an `AbstractVector`.
+   Then `length(B)` generators are defined whose names derived from a combination
+   of `A` and the respective element of `B`.
+   For example `:x => 1:3` defines three generators `x[1], x[2], x[3]`.
+3. A pair of the form `A => C`, where `A` is again a `VarName`, and `C` is
+   a tuple of ranges or v. For example `"a" => (1:2, 1:2)` defines four generators
+   `a[1, 1], a[2, 1], a[1, 2], a[2, 2]`.
 
-!!! warning "Note"
-    Variables named like the group generators are *not* created by this function.
+For the second and third type, optionally the `A` part can contain the
+placeholder `#` to modify where the indices are inserted. For example
+`"a#" => (1:2, 1:2)` defines four generators `a11, a21, a12, a22`.
+
+Also instead of a range, any vector can be used. For example `"#" => ([:x,:y], [:A, :B])`
+defines four generators `xA, yA, xB, yB`.
+
+In all variants, if the optional keyword argument `eltype` is given and
```

Comment:
The kwarg is missing from the second and third form's signatures at the top of the docstring